File: linked-list/insertarAlFinal/insertAtTheEnd.pas
// Procedimiento para insertar un elemento al final de una lista enlazada
procedure insertAtTheEnd(var actual, last: list; data: integer);
var
    aux: lista; // Variable para el nuevo nodo
begin
    new(aux); // Creamos un nuevo nodo
    aux^.data := data; // Asignamos el valor 'data' al campo 'data' del nodo
    aux^.next := NIL; // Establecemos el siguiente nodo como NIL (fin de la lista)

    // Si la lista no está vacía
    if actual <> Nil then
        last^.next := aux // Enlazamos el último nodo existente con el nuevo nodo
    else
        actual := aux; // Si la lista está vacía, el nuevo nodo es el primero

    last := aux; // El nuevo nodo se convierte en el último nodo
end;

File: linked-list/insertarAlFinal/insertAtTheEnd.ts
// Importamos la definición de la lista enlazada y el nodo desde '../type'
import { LinkedList, Node } from '../type';

// Clase que extiende la lista enlazada con un método para insertar al final
export class LinkedListWithInsertE extends LinkedList {
  insertAtTheEnd(data: number): void {
    const newNode = new Node(data); // Creamos un nuevo nodo con el valor 'data'

    if (!this.head)
      this.head = newNode; // Si la lista está vacía, el nuevo nodo es la cabeza
    else {
      let current = this.head; // Comenzamos desde la cabeza de la lista

      // Recorremos la lista hasta llegar al último nodo
      while (current.next) {
        current = current.next; // Avanzamos al siguiente nodo
      }

      current.next = newNode; // Enlazamos el último nodo existente con el nuevo nodo
    }
  }
}
